server/chat-plugins/chatlog.ts

```diff
@@ -640,7 +646,7 @@ export const commands: ChatCommands = {
 		if (!target) return this.parse('/help searchlogs');
 		if (!search) return this.errorReply('Specify a query to search the logs for.');
 		let limitString;
-		if (/^[0-9]+$/.test(limit)) {
+		if (/^[ 0-9]+$/.test(limit)) {
```

Comment:
Room IDs can't contain spaces; you should probably trim `limit` in its definition if this is an issue.